iree/compiler/Dialect/Util/IR/UtilOps.td

```diff
@@ -176,6 +176,37 @@ def Util_RangeExtentsOp : Util_PureOp<"range.extents", [
   let hasCanonicalizer = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// Alignment Arithmetic
+//===----------------------------------------------------------------------===//
+
+def Util_AlignOp : Util_PureOp<"align", [
+  AllTypesMatch<["value", "result"]>
+  ]> {
+  let summary = [{returns the (value + (alignment - 1)) & ~(alignment - 1)}];
+  let description = [{
+    Aligns |value| up to the given power-of-two |alignment| if required.
```

Comment:
may want a verifier checking that `alignment` is a power of two (if that is required)

Reply:
Per [discord discussion](https://discord.com/channels/689900678990135345/689906000043573354/905178436023238697) we'll emit this check for now since this is always a runtime value.